### ISSUE TITLE
Create your own VVC for UVVM

### ISSUE THREAD

**Issue by Ahmad-Zaklouta** [VHDL/news#7]
```yaml
ref: http://que.no/index.php/2017/07/03/uvvm-tutorial/
tags: [verification, simulation, UVVM]
```

**Comment by GlenNicholls** [comment]
Do you have any examples of using VUnit with UVVM?

**Comment by Ahmad-Zaklouta** [comment]
@GlenNicholls, unfortunately, no and I have looked but couldn't find. However, when I was trying to decide what to learn (VUnit, UVVM) I found that VUnit is much faster to learn. This example requires a lot of work to make it work.

**Comment by GlenNicholls** [comment]
@Ahmad-Zaklouta Yeah, I used both VUnit and UVVM before settling on VUnit. However, I found UVVM frustrating to learn so I didn't get very far. I understand the framework a decent amount, but having access to all of their VVC's in VUnit would be nice simply because VUnit doesn't have as many verification components. Not a huge deal, but I'm sure this is a deal-breaker for some people. The other example I see with using VUnit and UVVM is someone would be able to keep all of their UVVM frameworks, but with a modern test runner for CI.

**Comment by JimLewis** [comment]
@Ahmad-Zaklouta  OSVVM is also a nice alternative to UVVM and it plays nice with VUnit.   OSVVM utility library is a superset of what UVVM provides - in particular, UVVM currently provides the OSVVM Randomization and Functional Coverage packages in their release.    

Like VUnit, we do not provide as many verification components as UVVM - at least not yet.   OTOH, with respect to the AXI models, OSVVM provides the missing functionality the others do not - specifically the ability to run each of the independent AXI interfaces independently - yet still coordinate them.

**Comment by LarsAsplund** [comment]
@Ahmad-Zaklouta Using VUnit with any other VHDL framework is pretty straight-forward:

* If that framework is setup to stop the simulation on error then you don't have to do anything special. Anything that stops a VUnit testbench before `test_runner_cleanup` will be a failed test.
* If that framework may complete without stopping on errors, for example if you have a "stop level" set to 3 and only have one error, then you have to read that framework's error counter before just before calling `test_runner_cleanup`. Something like

``` vhdl
check_equal(function_to_read_error_counter, 0, result("for error counter"));
```
Give it a try and see what happens.

**Comment by Ahmad-Zaklouta** [comment]
@LarsAsplund I am actually using VUnit only at work and just playing with UVVM. For me, I feel that I got the structure and the idea of VUnit. For UVVM, I am still struggling to get the testbench structure. I feel my way is to focus more on learning VUnit with OSVVM because making this example work took so much time.
